### ISSUE TITLE
Should be able to delete shelf without emptying it first

### ISSUE THREAD

**Issue by placoderm** [bookwyrm-social/bookwyrm#2486]
**Is your feature request related to a problem? Please describe.**
Currently it is only possible to delete a shelf if it is empty. It's true that in the real world if you want to throw away a book shelf then you probably want to take the books of first. But since the shelves in this situation are not actual shelves, then that shouldn't matter.

**Describe the solution you'd like**
Currently the "Delete Shelf" button only appears when the shelf is empty:
![image](https://user-images.githubusercontent.com/31570884/206698751-7e7d2e78-d77b-437d-9eda-51f47d46b4a7.png)
Since books don't have to be on a custom shelf, users should be able to delete a shelf and automatically all the books categorized as that shelf should have that category removed.

**Describe alternatives you've considered**
Well, you could remove books one by one. If https://github.com/bookwyrm-social/bookwyrm/issues/1763 was ever implemented then you could bulk remove the books from the shelf and then delete the shelf.

**Comment by marcusyoung** [comment]
You will need to shelve the books somewhere though (and presumably all in the same shelf for usability) as they won't be part of your books unless shelved. Unless you just want rid of all the books and the shelf?

**Comment by placoderm** [comment]
So, maybe my confusion comes from some ambiguity around shelves in general.

I'm talking about *custom* shelves. I understand that a book needs to be on one of the default shelves. So if I am currently reading a book and I add it to a custom shelf "Non-Fiction," does that actually move it from the "Currently Reading" shelf? If so, that makes absolutely no sense. A books read/to be read/reading status has zero to do about how I might organize my collection. When I "move" a book from one custom shelf to another it doesn't actually move it; it just adds it to the second custom shelf. 

Or am I missing something? 

Honestly it would be much clearer if there was something like "Status" that was only about whether or not a book had been read, etc. And then there could be "Categories" for actually organizing a collection. As I investigate the app, the concept of "Shelf" seems not so great.

**Comment by marcusyoung** [comment]
You can have books on custom shelves which aren't on any of the built-in shelves. E.g. I've got an "own" shelf which is (or will be) every book I own, but most of those I don't want in any of the built-in shelves. I tend to agree re having a distinction between status and shelf. I think that may have come up for discussion before.

**Comment by placoderm** [comment]
Yeah, that's strange then. A "simple" solution would be to have an "unshelved" shelf. Which of course is illogical. But so are shelves here anyway. 😊

**Comment by marcusyoung** [comment]
If you have a book on your non-fiction shelf and 'move' it to "currently reading" it will stay in your non-fiction shelf. If you do that and then go to the page for the book, you'll also see the two locations where you have it shelved listed.

**Comment by placoderm** [comment]
> If you have a book on your non-fiction shelf and 'move' it to "currently reading" it will stay in your non-fiction shelf. If you do that and then go to the page for the book, you'll also see the two locations where you have it shelved listed.

The opposite, however is not true. I "moved" a book that was on the currently reading shelf to the non-fiction shelf. In this case it did actually move it because it is no longer on the currently reading shelf. 

But what you say is true. Now if I "move" the book from the non-fiction shelf to "currently reading" it is on both shelves.

This seems very wrong to me. Is this really the expected behaviour? 

I read thorough all the open issues and I didn't see anything about this.

**Comment by marcusyoung** [comment]
Not sure how much of this is expected or not. The fact that multiple shelves are listed with a heading on a book page suggests that having a book on multiple shelves is an anticipated event. But other than that, I really don't know.

**Comment by placoderm** [comment]
Thanks for all your feedback. Hopefully others may chime in.
